dino/src/game/dinosaur/systems.rs

```diff
@@ -36,53 +39,84 @@ pub fn handle_jump(
                 dinosaur_transform.translation.y += dino_vertical_movement.speed;
                 dino_vertical_movement.speed += GRAVITY;
             }
-        } else {
-            if keyboard_input.just_pressed(KeyCode::Space) {
-                dino_vertical_movement.moving = true;
-                dinosaur_transform.translation.y += dino_vertical_movement.speed;
-                dino_vertical_movement.speed += GRAVITY;
-            }
+        } else if keyboard_input.just_pressed(KeyCode::Space) && !dino_down.is_down {
+            dino_vertical_movement.moving = true;
+            dinosaur_transform.translation.y += dino_vertical_movement.speed;
+            dino_vertical_movement.speed += GRAVITY;
         }
     }
 }
 
 pub fn handle_collision(
     mut commands: Commands,
-    dinosaur_query: Query<&Transform, With<Dinosaur>>,
-    obstacle_query: Query<&Transform, With<Obstacle>>,
+    dinosaur_query: Query<(&Handle<Image>, &Transform), With<Dinosaur>>,
+    obstacle_query: Query<(&Handle<Image>, &Transform), With<Obstacle>>,
     mut game_over_event_writer: EventWriter<EndGame>,
     score: Res<Score>,
+    assets: Res<Assets<Image>>,
 ) {
-    if let Ok(dinosaur_transform) = dinosaur_query.get_single() {
-        for obstacle_transform in obstacle_query.iter() {
+    if let Ok((dinosaur_image_handle, dinosaur_transform)) = dinosaur_query.get_single() {
+        for (obstacle_image_handle, obstacle_transform) in obstacle_query.iter() {
+            let dinosaur_dimensions: Vec2 = assets.get(dinosaur_image_handle).unwrap().size();
+            let obstacle_dimensions: Vec2 = assets.get(obstacle_image_handle).unwrap().size();
```

Comment:
```suggestion
            let dinosaur_dimensions = assets.get(dinosaur_image_handle).unwrap().size();
            let obstacle_dimensions = assets.get(obstacle_image_handle).unwrap().size();
```
Não anote tipos que podem ser inferidos